### PR TITLE
Implement ExclusiveJets & EeKtAlgorithm 

### PR DESCRIPTION
**Pull request by Bastiantheone**
Solves #65 & #63 
I tried to model the exclusive jets after the c++ version

### CODE REVIEW

File: fastjet/clustersequence.go [hunk 1]
```diff
@@ -69,8 +72,63 @@ func NewClusterSequence(jets []Jet, def JetDefinition) (*ClusterSequence, error)
 	return cs, err
 }
 
+//returns the number of exclusive jets that would have been obtained
```
Comment (sbinet) [inline fastjet/clustersequence.go]: the Go style recommends to document functions like so:
```go
// NrExclusiveJets returns the number of ...
func (cs *ClusterSequence) NrExclusiveJets(...) { ...
```
also, `NrExclusiveJets` doesn't look too Go-ish.
see:
```
$> go doc flag.NArg 
func NArg() int
    NArg is the number of arguments remaining after flags have been processed.

$> go doc reflect.Type.NumField
// NumField returns a struct type's field count.
// It panics if the type's Kind is not Struct.
func NumField() int
```

so, well, even though the standard library (itself) doesn't follow a consistent style, `NrFoo` isn't part of that set of inconsistent options :)

I would go with `NumExclusiveJets`.

**Comment by sbinet** [comment]
ah! one more thing.

please add in your commit message:

```
Fixes go-hep/hep#63
Fixes go-hep/hep#65
```

so github can automatically close these issues when the CL will be merged.

File: fastjet/clustersequence.go [hunk 2]
```diff
@@ -69,8 +72,63 @@ func NewClusterSequence(jets []Jet, def JetDefinition) (*ClusterSequence, error)
...
+		err = errors.New("2*initial nr jets != size of history -- this endagers internal assumptions!")
+		panic(err)
+	}
+
+	//now go forwards and reconstitute the jets that we have --
+	//basically for any history element, see if the parent jets to
+	//which it refers were created before the stopping point -- if they
+	//were then add them to the list, otherwise they are subsequent
+	//recombinations of the jets that we are looking for
+	ljets := make([]Jet, 0)
```
Comment (Bastiantheone) [inline fastjet/clustersequence.go]: In the C++ version it checks at the end if ljets is the same size as the min of initn and njets. Would it be better to make a slice with that capacity: ljets := make([]Jet,0, imin(njets,initn)) ?

Comment (sbinet) [inline fastjet/clustersequence.go]: yes, if a good guess for the capacity can be provided, then we should file it.
that's that much less re-allocations to perform at runtime.

**Comment by Bastiantheone** [comment]
All changed up :)  I wouldn't mind implementing another one

**Comment by sbinet** [comment]
thanks a lot!

**Comment by sbinet** [comment]
next would be to work on the jet areas (p36 of http://fastjet.fr/repo/fastjet-doc-3.2.1.pdf) -- which would be a nice segway into voronoi diagrams and delaunay triangulations.

(I am out of town until Monday, though)

**Comment by Bastiantheone** [comment]
Sounds good

**Pull request merged**